cg/apps/balsamic/fastq.py

```diff
@@ -0,0 +1,150 @@
+# -*- coding: utf-8 -*-
+"""
+This module handles concatenation of balsamic fastq files.
+
+Classes:
+    FastqFileNameCreator: Creates valid balsamic filenames
+    FastqFileConcatenator: Handles file concatenation
+    FastqHandler: Handles fastq file linking
```

Comment:
Frankly: this is really great coding and a nice example of the single responsibility principle!

Reply:
Thank, I divided the code like that to avoid testing private functions which is a bit awkward. The tests shouldn't care about the implementation, just ensure that the promised functionality is present :)